app/grandchallenge/cases/models.py

```diff
@@ -842,6 +844,26 @@ def save(self, *args, **kwargs):
             )
 
 
+# Fixed placeholders for DICOM elements passed to Deidentifier
+# Important: Do not change as long as HealthImaging is used as storage backend!
+STUDY_DATE = "20000101"
+ACCESSION_NUMBER = ""
+PATIENT_ID = "ANONYMOUS"
+STUDY_ID = ""
+SERIES_NUMBER = "0"
+
+
+def generate_dicom_id_suffix(pk: str, suffix_type: str) -> int:
+    """
+    Generates a unique numerical suffix for a DICOM UID based on the primary key
+    and a string to differentiate the type (e.g., 'study' or 'series').
+    """
+    seed = f"{pk}-{suffix_type}"
+
+    digest = hashlib.sha512(seed.encode("utf8")).digest()
+    return int.from_bytes(digest[:14])
+
```

Comment:
Having tried this out, I actually think that relying on the pk will cause issues down the line. For that to work, a new model instance needs to be saved first with a default empty string. The DB uniqueness constraint will prevent saving other instances at the same time then. So with multiple concurrent DICOM uploads at the same time, we'd get lots of `IntegrityErrors`. 

I think we will need to create a UUID before the first save and pass that to `generate_dicom_id_suffix` to populate the two fields before calling `super().save()`.

Reply:
Why would it default to an empty string? The pk UUID is automatically generated on initialisation, so the suffix can be set pre-save.

With 32 digits using 10 characters per position the number of unique strings is going to be 10^36. I really don't see how collisions will be a problem with such a space.

Reply:
That is true when using auto field pks, in that case the pk comes from the database. For UUIDs they are generated in Python.